### PR TITLE
Add Idiom to MauiContext

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Add `Idiom` to MauiContext

Implements #1305

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

File: src/Core/src/Platform/Android/MauiContext.cs [hunk 1]
```diff
@@ -19,26 +25,78 @@ public MauiContext(IServiceProvider services, Context context) : this(context)
...
+		public Context? Context =>
+			_context.TryGetTarget(out Context? context) ? context : null;
+
+		public IServiceProvider Services =>
+			_services ?? throw new InvalidOperationException($"No service provider was specified during construction.");
+
+		public IMauiHandlersServiceProvider Handlers =>
+			_mauiHandlersServiceProvider ?? throw new InvalidOperationException($"No service provider was specified during construction.");
+
+		public TargetIdiom Idiom
```
Comment (AmrAlSayed0) [inline src/Core/src/Platform/Android/MauiContext.cs:40]: Does this really need to be reimplemented? Xamarin.Essentials already have an implementation for it. https://docs.microsoft.com/en-us/xamarin/essentials/device-information?tabs=ios

Comment (AmrAlSayed0) [reply src/Core/src/Platform/Android/MauiContext.cs:40]: Here https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs#L31-L73

Comment (jsuarezruiz) [reply src/Core/src/Platform/Android/MauiContext.cs:40]: You are right, must use Essentials as specified in Spec https://github.com/dotnet/maui/issues/1305

**Comment by jsuarezruiz** [comment]
After a team meeting, we are going to do this in a different way.

**Pull request closed without merging**